packages/utils/src/sha256.cairo

```diff
@@ -4,347 +4,372 @@
 
 //! Helpers for calculating double SHA256 hash digest.
 
-use core::num::traits::{Bounded, OverflowingAdd};
-
-/// Cairo implementation of the corelib `compute_sha256_byte_array` function.
-pub fn compute_sha256_byte_array(arr: @ByteArray) -> [u32; 8] {
-    let mut word_arr = array![];
-    let len = arr.len();
-    let rem = len % 4;
-    let mut index = 0;
-    let rounded_len = len - rem;
-    while index != rounded_len {
-        let word = arr.at(index + 3).unwrap().into()
-            + arr.at(index + 2).unwrap().into() * 0x100
-            + arr.at(index + 1).unwrap().into() * 0x10000
-            + arr.at(index).unwrap().into() * 0x1000000;
-        word_arr.append(word);
-        index = index + 4;
-    };
-    let last = match rem {
-        0 => 0,
-        1 => arr.at(len - 1).unwrap().into(),
-        2 => arr.at(len - 1).unwrap().into() + arr.at(len - 2).unwrap().into() * 0x100,
-        _ => arr.at(len - 1).unwrap().into()
-            + arr.at(len - 2).unwrap().into() * 0x100
-            + arr.at(len - 3).unwrap().into() * 0x10000,
-    };
-
-    compute_sha256_u32_array(word_arr, last, rem.into())
-}
-
-/// Cairo implementation of the corelib `compute_sha256_u32_array` function.
-pub fn compute_sha256_u32_array(
-    mut input: Array<u32>, last_input_word: u32, last_input_num_bytes: u32
-    ) -> [
-    u32
-; 8] {
-    add_sha256_padding(ref input, last_input_word, last_input_num_bytes);
+mod sha256 {
```

Comment:
Last time we discussed if this should be feature enabled (i'd also do that at the lib.cairo level rather than here tbh)

Reply:
Moved re declaration to lib.
For multi package project features does not work yet.